### PR TITLE
Enable lsst metadetect to read mcal types in config

### PR DESCRIPTION
**Pull request by zhouconghao**
Hi!

Currently, mdet only uses the DEFAULT_TYPES instead of reading the mcal types in the config file. In cluster shear calibration R11 and R22 are needed.

### CODE REVIEW

**Comment by esheldon** [comment]
Thanks for the fix.  Can you please also add a unit test that ensures the requested types are returned?

**Comment by zhouconghao** [comment]
@esheldon Done! Added a test in `test_lsst_metadetect.py`. There are two failures that are not related to metacal types. Happy to take a look at them later.

File: metadetect/lsst/metadetect.py [hunk 1]
```diff
@@ -74,8 +74,13 @@ def run_metadetect(
 
     fitter = get_fitter(config, rng=rng)
 
+    if 'types' in config['metacal']:
+        metacal_types = config['metacal']['types']
+    else:
+        metacal_types = None
```
Comment (beckermr) [inline metadetect/lsst/metadetect.py:80]:
```suggestion
    metacal_types = config['metacal'].get('types', None)
```

**Comment by beckermr** [comment]
This PR needs to have main merged into it.

**Pull request merged**